### PR TITLE
feat(plugin): RPC eth_sendTransaction method added and bugs fixed

### PR DESCRIPTION
**Pull request by OlehPerehuda**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)

      window.ethereum.request({
        method: 'eth_sendTransaction',
        params: [
          {
          to: '0x0d7dB08D75679bbE90b7B1eDfB8BE3a16897Ee77',
          value: utils.parseEther(0.1)._hex,
         gasLimit: 21000
        ]
      });

### CODE REVIEW

**Review by vladdezhniuk — APPROVED** [review]

**Comment by vladdezhniuk** [comment]
LGTM!!!!

**Comment by vladdezhniuk** [comment]
AWESOME

**Comment by vladdezhniuk** [comment]
Glek!

File: packages/blockchain-wallet-v4-frontend/src/data/components/plugin/sagas.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default ({ api }: { api: APIType }) => {
       yield put(A.setWallet(wallet))
       return wallet
     } catch (e) {
-      throw new Error(WALLET_SIGNER_ERR)
+      throw new Error(`Failed to get address. ${e}`)
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/data/components/plugin/sagas.ts]: Failed to get wallet

File: packages/blockchain-wallet-v4-frontend/src/plugin/internal/background.ts [hunk 2]
```diff
@@ -58,13 +60,28 @@ chrome.runtime.onConnect.addListener(async (port: chrome.runtime.Port) => {
             })
           }
           break
+        case SupportedRPCMethods.Send:
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/plugin/internal/background.ts]: SendTransaction

File: packages/blockchain-wallet-v4-frontend/src/plugin/provider/utils.ts [hunk 3]
```diff
@@ -21,3 +30,40 @@ export enum SupportedRPCMethods {
...
+
+export const isTransactionParametersType = (transactionsParameter: any) =>
+  !!(transactionsParameter.to && transactionsParameter.value)
+
+export const isValidEthAddress = (address: string) => /^0x[a-fA-F0-9]{40}$/.test(address)
+
+export const validateSendTransactionRequestParams = (params) => {
+  ;(params as providers.TransactionRequest[]).forEach((param) => {
+    if (!isTransactionParametersType(param)) {
+      throw ethErrors.provider.custom({
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/plugin/provider/utils.ts]: use rpc errors here

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/SendTransaction/index.tsx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        params.get('to') || '',
+        params.get('from') || '',
+        params.get('nonce') || '',
+        params.get('gasLimit') || 0,
+        params.get('gasPrice') || '',
+        params.get('data') || '',
+        params.get('value') || 0,
+        Number(params.get('chainId')) || 0
+      )))
+    })()
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/SendTransaction/index.tsx]: implement separate hook/function for this screen and signTransaction screen

**Comment by TarasPotsikaylo** [comment]
:wastebasket:  :adhesive_bandage:

File: packages/blockchain-wallet-v4-frontend/src/data/components/plugin/sagas.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ import { askSecondPasswordEnhancer } from 'services/sagas'
 
 import { actions as signerActions } from './slice'
 
-const logLocation = 'auth/sagas'
+const logLocation = 'components/nfts/sagas'
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/data/components/plugin/sagas.ts]: what is this change?

**Review by VitaliiShpital — APPROVED** [review]

**Pull request merged**